src/main/java/se/sundsvall/billingdatacollector/integration/opene/mapper/kundfakturaformular/KundfakturaformularMapper.java

```diff
@@ -0,0 +1,235 @@
+						.withActivity(getLeadingDigits(result.aktivitetskonto()))
+						.withCounterpart(getInternalMotpartNumbers(customerId))
+						.build())
+					.build()))
+				.build())
+			.build();
+
+		return BillingRecordWrapper.builder()
+			.withBillingRecord(billingRecord)
+			.build();
+	}
+
+	BillingRecordWrapper mapToExternalBillingRecord(final byte[] xml) {
+		var result = extractValue(xml, ExternFaktura.class);
+
+		var billingRecord = BillingRecord.builder()
+			.withCategory(CATEGORY)
+			.withStatus(Status.APPROVED)
+			.withType(Type.EXTERNAL)
+			.withRecipient(Recipient.builder()
+				.withFirstName(result.fornamn())
+				.withLastName(result.efternamn())
+				.withAddressDetails(AddressDetails.builder()
+					.withStreet(result.adress())
+					.withPostalCode(result.postnummer())
+					.withCity(result.ort())
+					.build())
+				.build())
+			.withInvoice(Invoice.builder()
+				.withInvoiceRows(List.of(InvoiceRow.builder()
+					.withDescriptions(List.of(result.fakturaText()))
+					.withQuantity(result.antal())
+					.withCostPerUnit(convertStringToFloat(result.aPris()))
+					.withVatCode(result.momssats())
+					.withTotalAmount(convertStringToFloat(result.summering()))
+					.withAccountInformation(AccountInformation.builder()
+						.withCostCenter(getLeadingDigits(result.ansvar()))
+						.withSubaccount(getLeadingDigits(result.underkonto()))
+						.withDepartment(getLeadingDigits(result.verksamhet()))
+						.withActivity(getLeadingDigits(result.aktivitetskonto()))
+						.withArticle(result.objektkonto())
+						.withCounterpart(getExternalMotpartNumbers(result.motpartNamn()))
+						.build())
+					.build()))
+				.build())
+			.build();
+
+		return BillingRecordWrapper.builder()
+			.withBillingRecord(billingRecord)
+			.withLegalId(result.personnummer())
+			.build();
+	}
+
+	//////
+	// Lots of converters and stuff down here.
+	// Move to a separate class if we need them in more places in the future.
+	//////
+
+	/**
+	 * Converts a string to a float.
+	 * e.g. "123,45 SEK" will be converted to Float: 123.45
+	 * @param stringToConvert The string to convert
+	 * @return The string converted to a float
+	 */
+	Float convertStringToFloat(String stringToConvert) {
+		try {
+			var withRemovedCurrency = removeCurrencyFromString(stringToConvert);
+			var withReplacedCommas = replaceCommasInCurrencyString(withRemovedCurrency);
+			return Float.parseFloat(withReplacedCommas);
+		} catch (Exception e) {
+			//Catch all the errors..
+			throw Problem.builder()
+				.withTitle(String.format(STRING_TO_FLOAT_ERROR, stringToConvert))
+				.withStatus(INTERNAL_SERVER_ERROR)
+				.build();
+		}
+	}
+
+	/**
+	 * Replaces commas with dots in a currency string to be able to parse it to a Float
+	 * @param stringToParse The string to parse
+	 * @return The string with commas replaced with dots (if any).
+	 */
+	String replaceCommasInCurrencyString(String stringToParse) {
+		return stringToParse.replace(",", ".");
```

Comment:
Comment applicable on all this shit:

Do everything null safe?
Example:

	String removeCurrencyFromString(String stringToParse) {
	        return Optional.ofNullable(stringToParse)
                       .map(string -> string.replaceAll("[^0-9.,]", ""))
                       .orElse(null);
	}